### PR TITLE
Refactor FXIOS-6770 [v116] Improve dependency injection - EnhancedTrackingProtectionMenuVM

### PR DESCRIPTION
**Pull request by mauriciomaniglia**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6770)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15069)

### Description
- Decouple the `Tab` dependency from EnhancedTrackingProtectionMenuVM and inject the URL, displayTitle and BlockerStatus instead.
- Decouple the `Profile` dependency and inject only the boolean that we need for `globalETPIsEnabled`. 
- Create new closure `onToggleSiteSafelistStatus` to be called inside the function `toggleSiteSafelistStatus`, that way this class doesn't need to reload the `Tab` and delegates back to BrowserViewController.
- Create unit tests for EnhancedTrackingProtectionMenuVM. 

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [x] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift [hunk 1]
```diff
@@ -109,8 +109,16 @@ extension BrowserViewController: URLBarDelegate {
...
+        if let tab = self.tabManager.selectedTab,
+            let url = tab.url,
+            let contentBlocker = tab.contentBlocker,
+            let webView = tab.webView {
+            let etpViewModel = EnhancedTrackingProtectionMenuVM(
+                url: url,
+                displayTitle: tab.displayTitle,
+                connectionSecure: webView.hasOnlySecureContent,
+                globalETPIsEnabled: FirefoxTabContentBlocker.isTrackingProtectionEnabled(prefs: profile.prefs),
+                contentBlockerStatus: contentBlocker.status)
```
Comment (OrlaM) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift:121]: Nit: This function would read much nicer if this was a guard.

**Review by OrlaM — APPROVED** [review]
Thanks! I'll just run it on the CI and then I'll merge it.

**Comment by lmarceau** [comment]
@mauriciomaniglia There were some failing tests on the CI, could you look into those?
Screenshot 2023-06-21 at 1 34 38 PM

**Comment by mauriciomaniglia** [comment]
> @mauriciomaniglia There were some failing tests on the CI, could you look into those?
> Screenshot 2023-06-21 at 1 34 38 PM

@lmarceau The CI doesn't load the property `accessibilityIdentifier` for some reason. I ran the tests locally, and it's working. I was thinking maybe, instead of the ViewModel delivering the UIImage, we can send the image name. That way is easier to test, and we can decouple the ViewModel from UIKit and let the ViewController load the image. But then we will need to duplicate some logic because we render the picture differently depending on some conditions. We will have the same logic in both ViewModel and ViewController. I think for now, I will leave it as it is and remove the unit test for this specific case.

**Comment by lmarceau** [comment]
Build is green, merging! Thanks @mauriciomaniglia for your contribution!

**Pull request merged**